docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -597,7 +631,8 @@ Instead, Quarkus will pass `null` for the argument `library`.
 That gives you option to reuse your custom Permission when the method argument (like `library`) is optional.
 <2> Argument `library` will be passed to the `LibraryPermission` constructor as the `LibraryService#updateLibrary` method is not an endpoint.
 
-Currently, there is only one way to add permissions, and that is xref:security-customization.adoc#security-identity-customization[Security Identity Customization].
+The only one way to add custom `java.security.Permission` that accepts additional arguments like `LibraryPermission` is
```

Comment:
This point should probably be clarified, as it may not be clear to users that when using HTTP security policy configuration, the method parameters have not been made available yet

Reply:
Well that's actually not a problem (remember, there are 2 permissions - one holds `SecurityIdentity`, one is created for annotation instance, then there is check securityIdentityPermission implies annotationInstancePermission). The reason why you need to use augmentor is that if your permission accepts more than name and actions, we don't know how to supply them (e.g. if it accepts class instance, how will you suply it via configuration properties).. So I made it simpler this way. Anyway, if you are implementing permission yourself, you are in charge and you can assing identity different permission than to `@PermissionAllowed` annotation. Something like `LibraryPermissionCheckingPermission` that is capable to imply `LibraryPermission` (remember, it can do what you want...) :-). 

Even after this explanation, do you still require to make this sentence more explanative @sberyozkin ? The sentence is not incorrect.

btw. I plan dedicated documentation PR as I got impression that LibraryPermission is not relatable to some developers (that's my conclusion from dev group).

Reply:
I should probably add that only permission that is instantiated for annotation instance holds method arguments (after all, you can have one `SecurityIdentity` that invokes dozen methods secured with `@PermissionsAllowed`, it could be mess otherwise).

Reply:
Hey @sberyozkin , today I thought about it again and re-read the text. IMO it's common sense that you can't create Permission class which constructor accepts objects that can't be provided as strings (configuration properties) other way than via `SecurityIdentityAugmentor`. WDYT?

Reply:
Hi Michal, let me clarify something,  so with this PR, we allow users to avoid having to augment the identity with the custom identity augmentor and continue relying on `@PermissionAllowed` ? Just would like to make sure we are on the same line before commenting further,  thanks

Reply:
@sberyozkin ,yes please. This PR does one thing - when HTTP Security Policy grants access to path, this PR allows you to grant `SecurityIdentity` permissions to certain role (admin etc.). You can see in documentation that now, `CRUDResource` doesn't need augmentor if your `SecurityIdentity` has roles.

side note: every time you ask such question I very much doubt my documentation skills :-D I'll have to work on that.

Reply:
@michalvavrik Thanks for the confirmation

Reply:
@michalvavrik So, when we talk about custom permissions which accept extra parameters, we are talking about the checkers, right ? I.e the checker permission will get an extra parameter passed to it, and then this checker will use this extra parameter to decide if it implies the given permission. 
If it is the case and given that `@PermissionAllowed` is also used which is where it is decided which extra parameters must be passed, why do you have this note about the security identity customization - to me it implies the manual security identity augmentation is needed, but in this PR the augmentation is done automatically

Reply:
I also believe it is important to document properly adding permissions via augmentor as it allows you to not rely on role-based authorization (if needed). However I can see your point now. Thank you

Reply:
I believe if docs is unclear to you who knows Quarkus Security well, than it's going to be way worse for users to understand. TBH I'm lost on weak points of current examples (it is clear to me) and feedback is welcome.

Reply:
@michalvavrik Sorry, your PR is great, it is, what I find though that the docs focus on some technical aspects which, IMHO, might not be relevant to what the actual PR is trying to achieve.
What I understand is that a Permission instance which is added in SecurityAugmentor does not have access to the actual Resteasy request parameters , the Permission instance added there is meant to be passed to the checker's `implies` method, and it is that that `checker` (and not this Permission instance added in the augmentor) that will need to, if it wants to, accept the extra request parameters so that it can use them to do some extra checks.

If users really want to pass the extra parameters to  Permission instance when adding them in the augmentor, well, they can do it, but it is not really relevant to what your PRs are making happening, you show the examples where a Media Library is added as an extra param in the augmentor, but this parameter is not related to the current request..., the same MediaLibrary can be injected as a bean into the checker CDI bean instead.

And this is why I'm getting confused, when I see a highlight in this PR talking about the augmentors, with the PR's focus on making the use of augmentors unnecessary.

I'm sorry if I make it unnecessarily complicated, this review, like I said, your PR is great, and I'll approve shortly, I have a couple of other minor doc suggestions in mind, will do them a bit later

Reply:
> What I understand is that a Permission instance which is added in SecurityAugmentor does not have access to the actual Resteasy request parameters , the Permission instance added there is meant to be passed to the checker's `implies` method, and it is that that `checker` (and not this Permission instance added in the augmentor) that will need to, if it wants to, accept the extra request parameters so that it can use them to do some extra checks.

That's correct. 

> If users really want to pass the extra parameters to Permission instance when adding them in the augmentor, well, they can do it, but it is not really relevant to what your PRs are making happening, you show the examples where a Media Library is added as an extra param in the augmentor, 

That's also correct.

> but this parameter is not related to the current request..., the same MediaLibrary can be injected as a bean into the checker CDI bean instead.

Technically it is also correct, but I believe we don't agree on why is the `MediaLibrary` in the permission added by the augmentor. My intention was to show user that they can create `LibraryPermission` that will use `LibraryPermission#isParentLibraryOf` and behave differently if user has `MediaLibrary`. I am considering to rewriting it to something Steph suggested (related to Github, I though about PullRequestService or something). It needs to be done in separate PR.

> And this is why I'm getting confused, when I see a highlight in this PR talking about the augmentors, with the PR's focus on making the use of augmentors unnecessary.

I need to adjust part about augmentor to make it clear there is easier way to do that.

Sure, let me know docs adjustments you have in mind, thank you.